core/src/graph/updates.rs

```diff
@@ -83,14 +83,14 @@ impl UpdateTracker {
 	#[log_result_err(Level::Info)]
 	pub fn register_updates(
 		&mut self,
-		events: &[UpdateEvent],
+		mut events: Vec<UpdateEvent>,
```

Comment:
I don't think this `mut` is necessary.  you can just use `into_iter() ` on the vec and it will give you the item

Reply:
Even if I'm using `.drain(..)` below?

Reply:
fixed; changed `.drain(..)` to `.into_iter()` and removed `mut`.